### DISCUSSION TITLE
Visual Testing

### DISCUSSION THREAD

**Discussion by preschian** [kodadot/nft-gallery#4803] [General]
e2e test helps me to catch some unintended bugs like this one https://github.com/kodadot/nft-gallery/pull/4782#issuecomment-1398588295. but, still not able to catch the visual/CSS part https://github.com/kodadot/nft-gallery/pull/4782#issuecomment-1399224852. need to set up additional config for visual testing. maybe it will help us to keep matched with the design

visual testing tools:
- https://github.com/marketplace/argos-ci
- https://percy.io/
- https://docs.cypress.io/guides/tooling/visual-testing#Tooling

cc @kodadot/code-review-guild

**Comment by vikiival** [comment]
Wow, that would be really nice!

**Comment by yangwao** [comment]
is this what @exezbcz doing manualy per PR? :D 

I mean dream state would be if that visual testing could follow figma somehow?

**Reply by preschian** [reply]
> is this what @exezbcz doing manualy per PR? :D

XD it's time to @exezbcz had a buddy review

**Comment by yangwao** [comment]
So, how we can engage with this further? I guess best desire would be set some % close to Figma reality and then put label on PR I guess.
